connect_four/ai.rb

```diff
@@ -0,0 +1,38 @@
+require './board.rb'
+
+class AI
+  attr_reader :player_id
+  attr_accessor :game_board
+
+  def initialize(player_id, game_board)
+    @player_id = player_id
+    @game_board = game_board
+  end
+
+  def pick_column
+    rational_choice = pick_rationally
+    if rational_choice
+      rational_choice
+    else
+      pick_randomly
+    end
+  end
+
+  private
+
+  def pick_rationally
+    (0..6).each do |column|
+      board_copy = Board.new
+      board_copy.board = game_board.board.map { |column| column.dup }
```

Comment:
I came across some of this with my code, actually may have a game_board.board too. I KNEW I didn't want board.board. Sometimes that's the best you can do to avoid confusion.